src/Entity/WebHook/WebHook.php

```diff
@@ -27,76 +27,83 @@ class WebHook
      *
      * @ORM\Column(length=64)
      *
-     * @JMS\Groups({"api"})
+     * @Assert\NotBlank
      */
     private $event;
 
     /**
-     * @var Collection|Callback[]
+     * This property is needed to manage to have the unique constraint "web_hook_event_client_id_unique" with Doctrine ORM.
      *
-     * @ORM\OneToMany(targetEntity="AppBundle\Entity\WebHook\Callback", mappedBy="webHook", cascade={"all"}, orphanRemoval=true)
+     * @var Client
      *
-     * @Assert\Valid
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\OAuth\Client", inversedBy="webHooks")
+     * @ORM\JoinColumn(nullable=false)
+     *
+     * @Assert\NotNull
+     */
+    private $client;
+
+    /**
+     * @var string[]
+     *
+     * @ORM\Column(type="json")
+     *
+     * @Assert\Count(min=1, minMessage = "Veuillez spécifier au moins une url de callback.")
      */
     private $callbacks;
 
-    public function __construct(Event $event, $callbacks = [])
+    public function __construct(Client $client = null, Event $event = null, $callbacks = [])
     {
         $this->uuid = Uuid::uuid4();
-        $this->event = $event->getValue();
-        $this->callbacks = new ArrayCollection();
+        $this->client = $client;
+        $this->event = $event ? $event->getValue() : null;
 
+        $this->callbacks = [];
```

Comment:
do this on declaration of the property `private $callbacks = [];`

Reply:
What's the advantages of doing so ?

Reply:
I think it's more clearer and readable to use the construct for instantiating object / inject values and let the definition of scalar value on property declaration. 
```php
class Foo
{
    private $status;
    private $step;
    private $children;
    private $createdAt;

    function __construct()
    {
        $this->status = 'DEFAULT STATUS';
        $this->step = 1; // default step
        $this->children = [];
        $this->createdAt = new \DateTime;
    }
}
```

```php
class Foo
{
    private $status = 'DEFAULT STATUS';
    private $step = 1; // default step;
    private $children = [];
    private $createdAt;

    function __construct()
    {
        $this->createdAt = new \DateTime;
    }
}
```

Reply:
IMO, it's simpler to put everything in the constructor so that you know it's all in the constructor and you don't need to look for anything anywhere else.